helpers/kubernetes_cluster.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+class Clover
+  def kubernetes_cluster_post(name)
+    authorize("KubernetesCluster:create", @project.id)
+
+    required_parameters = ["name", "location", "version", "private_subnet_id", "cp_nodes", "worker_nodes"]
+    request_body_params = validate_request_params(required_parameters)
+
+    private_subnet = PrivateSubnet.from_ubid(request_body_params["private_subnet_id"])
+
+    unless private_subnet && private_subnet.location == @location
+      fail Validation::ValidationFailed.new({private_subnet_id: "Private subnet with the given id \"#{request_body_params["private_subnet_id"]}\" and the location \"#{@location}\" is not found"})
+    end
+
+    authorize("PrivateSubnet:edit", private_subnet.id)
+
+    kc = nil
+
+    DB.transaction do
+      kc = Prog::Kubernetes::KubernetesClusterNexus.assemble(
+        name: name,
+        version: request.params["version"],
+        private_subnet_id: private_subnet.id,
+        project_id: @project.id,
+        location: @location,
+        cp_node_count: request.params["cp_nodes"].to_i
+      ).subject
+
+      Prog::Kubernetes::KubernetesNodepoolNexus.assemble(
+        name: name + "-np",
+        node_count: request.params["worker_nodes"].to_i,
+        kubernetes_cluster_id: kc.id
+      )
+    end
+
+    flash["notice"] = "'#{name}' will be ready in a few minutes"
```

Comment:
You don't have to change things, but this can be inside the transaction block, so you don't need the earlier `kc = nil`.

Reply:
thanks, I moved them inside.

funny thing is, I vaguely remember moving those out of the transaction block due to another review comment, but I cannot find it :D